bugsnag-android-core/src/main/java/com/bugsnag/android/Journal.java

```diff
@@ -0,0 +1,170 @@
+package com.bugsnag.android;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.dslplatform.json.DslJson;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A journal represents a series of commands that will manipulate a document.
+ * Sometimes it's more efficient to store a base document and a list of modifications
+ * to me made to it rather than re-serialize the entire document after every modification.
+ */
+public class Journal {
+    @NonNull
+    public final List<Command> commands;
+
+    public Journal() {
+        commands = new LinkedList<>();
```

Comment:
Will this class be called from multiple threads? What is the strategy used to avoid [concurrent modification exceptions](https://docs.oracle.com/javase/7/docs/api/java/util/ConcurrentModificationException.html) if so?

Reply:
There will be an outer class that manages the shared memory region between java and native side. This will already have to handle concurrency, so it will protect the journal as well since it has to add entries to the journal anyway.